tests/0124-openssl_invalid_engine.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * librdkafka - Apache Kafka C library
+ *
+ * Copyright (c) 2021, Magnus Edenhill
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "test.h"
+
+int main_0124_openssl_invalid_engine (int argc, char **argv) {
+#if WITH_SSL && OPENSSL_VERSION_NUMBER >= 0x10100000
```

Comment:
Don't think we can rely on the OPENSSL defines to be available here. I think a simpler alternative is to skip the test if
conf_set("ssl.engine.location") returns UNKNOWN. E.g.:
```c
 rd_kafka_conf_res_t res;
 ...
        res = rd_kafka_conf_set(conf, "ssl.engine.location", "invalid_path", 
                              errstr, sizeof(errstr));
        if (res != RD_KAFKA_CONF_UNKNOWN) {
                  rd_kafka_destroy(conf);
                  TEST_SKIP("%s\n", errstr);
                  return 0;
        } else if (res != RD_KAFKA_CONF_OK)
                 TEST_FAIL("%s", errstr);
....

```

Reply:
i think you mean
if (res **==** RD_KAFKA_CONF_UNKNOWN) {
                  rd_kafka_destroy(conf);
                  TEST_SKIP("%s\n", errstr);
                  return 0;